respy/method_of_simulated_moments.py

```diff
@@ -0,0 +1,316 @@
+"""Estimate models with the method of simulated moments (MSM).
+
+The method of simulated moments is developed by [1], [2], and [3] and an estimation
+technique where the distance between the moments of the actual data and the moments
+implied by the model parameters is minimized.
+
+References
+----------
+.. [1] McFadden, D. (1989). A method of simulated moments for estimation of discrete
+       response models without numerical integration. Econometrica: Journal of the
+       Econometric Society, 995-1026.
+.. [2] Lee, B. S., & Ingram, B. F. (1991). Simulation estimation of time-series models.
+       Journal of Econometrics, 47(2-3), 197-205.
+.. [3] Duffie, D., & Singleton, K. (1993). Simulated Moments Estimation of Markov Models
+       of Asset Prices. Econometrica, 61(4), 929-952.
+
+"""
+import copy
+import functools
+import itertools
+
+import numpy as np
+import pandas as pd
+
+from respy.simulate import get_simulate_func
+
+
+def get_msm_func(
```

Comment:
Just noticed that the user should be able to use the argument ``n_simulation_periods`` from ``simulate()``. It can be used to control the number of simulated periods in contrast to ``options["n_periods"]`` which defines for how many periods individuals maximize their expected utility. You could say, I want individuals to maximize utility until retirement, but I only need the first 20 years of simulated behavior.